aip/cloud/2510.md

```diff
@@ -67,6 +67,46 @@ considered PII and user data, but project numbers are not.
 Therefore, when an internal service calls an external Google APIs, it
 **should** use project numbers for making API requests.
 
+### Resource References
+
+Project IDs also appear in [resource names][], which are stored in other
+resources when they are referenced ([example][]).
+
+In these situations, the value of the resource reference should be preserved
+regardless if the user has sent a project ID or number to identify the main
+resource.
+
+For example, for an existing resource book:
+
+```proto
+book {
+  // project-id: "foo"
+  // project-number: 12345
+  name: "project/12345/books/game-of-tones"
+  shelf: "project/123456/shelves/top-row"
+}
```

Comment:
Do we express proto values in this form elsewhere?

Reply:
All the other examples that I could find are .proto representations of a message or RPC.

This is similar to the textproto representation found in Google's code base, since I couldn't find an example within AIPs.

Reply:
I could possibly use the capital `Book` in the prefix, which would kind of map to the proto representation of a message.

Reply:
resolved given accepting @shwoodard's PR and making a few follow-up adjustments.